Data/HashMap/Base.hs

```diff
@@ -745,6 +753,29 @@ map :: (v1 -> v2) -> HashMap k v1 -> HashMap k v2
 map f = mapWithKey (const f)
 {-# INLINE map #-}
 
+-- | /O(n)/. The function 'mapAccum' threads an accumulating
+-- argument through the map in some unspecified order of keys.
+mapAccum :: (a -> v1 -> (a,v2)) -> a -> HashMap k v1 -> (a,HashMap k v2)
+mapAccum f = mapAccumWithKey (\a' _ x' -> f a' x')
+
+-- | /O(n)/. The function 'mapAccumWithKey' threads an accumulating
```

Comment:
You need a lazy version and a strict version. The lazy one shouldn't evaluate the value thunk before putting it in the map.

There's also the issue of the strictness of the accumulator. We need to document it's strictness (i.e. is the accumulator treated as in `foldl` or in `foldl'`.)